src/main.cpp

```diff
@@ -427,6 +426,12 @@ static void setupWebServer()
         gWebServer.send(302, "text/plain", "");
     });
 
+    gWebServer.on("/change-partition", HTTP_GET, []() {
+        gWebServer.send(200, "text/plain", "Restart initiated!");
+        setAppPartition0Active();
+        ESP.restart();

```

Comment:
If you restart immediately, it might happen that the HTML 200 code will never be sent to the browser.